app/src/main/kotlin/com/wire/android/ui/calling/SharedCallingViewModel.kt

```diff
@@ -168,8 +178,8 @@ class SharedCallingViewModel @Inject constructor(
         }
     }
 
-    private suspend fun observeCallState() {
-        allCalls().collect { calls ->
+    private suspend fun initCallState(sharedFlow: SharedFlow<List<Call>>) {
+        sharedFlow.first { calls ->
```

Comment:
You are executing actions in the predicate function which should only decide if the element should be collected or not, so it would be better this way:
```
            sharedFlow.first().find { call ->
                call.conversationId == conversationId &&
                        call.status != CallStatus.CLOSED &&
                        call.status != CallStatus.MISSED
            }?.let { call ->
                callState = callState.copy(
                    callStatus = call.status,
                    callerName = call.callerName,
                    isCameraOn = call.isCameraOn,
                    isMuted = call.isMuted
                )
            }
```

Reply:
Or even maybe we could just pass `List<Call>` as a parameter of this function instead of `SharedFlow`